auth/passwords.hh

```diff
@@ -75,10 +76,18 @@ sstring generate_salt(RandomNumberEngine& g, scheme scheme) {
 
 ///
 /// Hash a password combined with an implementation-specific salt string.
+/// Deprecated in favor of `hash_with_salt_async`.
 ///
 /// \throws \ref std::system_error when an unexpected implementation-specific error occurs.
 ///
-sstring hash_with_salt(const sstring& pass, const sstring& salt);
+[[deprecated("Use hash_with_salt_async instead")]] sstring hash_with_salt(const sstring& pass, const sstring& salt);
```

Comment:
I'm not convinced the argument to deprecate this function instead of removing it is valid: I don't understand how it helps "reduce regression risk" if we keep this deprecated function to be used in half of the cases - but still not all of them.

I'm worried that it's even the opposite: Because in some of the cases (?) we still use the old function, it's possible that the new function has a bug but a test we have that might have caught this bug happens to be one that still uses the old function.